src/main/java/com/faforever/client/game/CreateGameController.java

```diff
@@ -522,10 +525,20 @@ public void onMapFilterButtonClicked() {
     }
   }
 
-  public void onMapPreviewImageClicked() {
-    Optional.ofNullable(mapListView.getSelectionModel())
-        .map(SelectionModel::getSelectedItem)
-        .map(MapVersionBean::getFolderName)
-        .ifPresent(mapName -> PopupUtil.showImagePopup(mapService.loadPreview(mapName, PreviewSize.LARGE)));
+  public void onMapPreviewImageClicked(MouseEvent mouseEvent) {
+    MouseButton button = mouseEvent.getButton();
+    if (button == MouseButton.PRIMARY) {
+      Optional.ofNullable(mapListView.getSelectionModel())
+          .map(SelectionModel::getSelectedItem)
+          .map(MapVersionBean::getFolderName)
+          .ifPresent(mapName -> PopupUtil.showImagePopup(mapService.loadPreview(mapName, PreviewSize.LARGE)));
+
+      // No other way to make same design without using StackPane class therefore we use this dirty hack
+    } else if (button == MouseButton.SECONDARY && mapNameLabel.getBoundsInParent().contains(mouseEvent.getX(), mouseEvent.getY())) {
+      contextMenuBuilder.newBuilder()
+          .addItem(CopyLabelMenuItem.class, mapNameLabel)
+          .build()
+          .show(mapNameLabel, mouseEvent.getScreenX(), mouseEvent.getScreenY());
+    }
```

Comment:
Rather than complicating the onMapPreviewImageClicked, add a new function and assign it to onContextMenuRequested of the stackPane perhaps